src/nupic/math/ArrayAlgo.hpp

```diff
@@ -3779,7 +3779,7 @@ namespace nupic {
       // n1 is the number floats we can process in parallel with xmm
       // n2 is the number of "stragglers" what we will have to do one by one ( < 4)
       nupic::Real32 count = 0;
-      long x_addr = (long) begin; // 8 bytes on 64 bits platforms
+      long long x_addr = (long long) begin; // 8 bytes on 64 bits platforms
```

Comment:
A unit test in the core for the FDR Spatial class is needed to test this, or testing through NuPIC. It is to overcome a 64 bit pointer truncation.